src/tfr_aruco/src/tfr_aruco_server.cpp

```diff
@@ -0,0 +1,134 @@
+#include "ros/ros.h"
+
+// aruco and ROS-openCV bindings
+#include <opencv2/aruco.hpp>
+#include <image_transport/image_transport.h>
+#include <cv_bridge/cv_bridge.h>
+#include <image_geometry/pinhole_camera_model.h>
+#include <sensor_msgs/image_encodings.h>
+
+#include <tfr_msgs/ArucoAction.h>
+#include <actionlib/server/simple_action_server.h>
+
+
+typedef actionlib::SimpleActionServer<tfr_msgs::ArucoAction> Server;
+
+// This is the method that will be called when a client makes use
+// of this server. The provided goal is the "input".
+void execute(const tfr_msgs::ArucoGoalConstPtr& goal, Server* server)
+{
+    cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_5X5_250);
```

Comment:
You do not check for preemption at the beginning of an action, this needs to be fixed.